stl/inc/span

```diff
@@ -0,0 +1,678 @@
+        return _Iter._Unwrapped();
+    }
+};
+#endif // _HAS_CXX20
+
+// STRUCT TEMPLATE _Span_extent_type
+template <size_t _Extent>
+struct _Span_extent_type {
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type([[maybe_unused]] const size_t _Size) noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Size == _Extent, "Mismatch between static extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept {
+        static_assert(_OtherExtent == _Extent || _OtherExtent == dynamic_extent,
+            "Mismatch between static extent and size of initializing data");
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_OtherExtent == dynamic_extent) {
+            _STL_VERIFY(_Ext._Size() == _Extent, "Mismatch between static extent and size of initializing data");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    _NODISCARD constexpr size_t size() const noexcept {
+        return _Extent;
+    }
+};
+
+template <>
+struct _Span_extent_type<dynamic_extent> {
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type(const size_t _Size) noexcept : _Mysize(_Size) {}
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext._Size()) {}
+
+    _NODISCARD constexpr size_t size() const noexcept {
+        return _Mysize;
+    }
+
+private:
+    size_t _Mysize{0};
+};
+
+// STRUCT TEMPLATE _Is_span
+template <class _Ty, size_t _Extent>
+class span;
+
+template <class>
+struct _Is_span : false_type {};
+
+template <class _Ty, size_t _Extent>
+struct _Is_span<span<_Ty, _Extent>> : true_type {};
+
+// STRUCT TEMPLATE _Is_std_array
+template <class>
+struct _Is_std_array : false_type {};
+
+template <class _Ty, size_t _Extent>
+struct _Is_std_array<array<_Ty, _Extent>> : true_type {};
+
+// STRUCT TEMPLATE _Is_span_convertible_range
+template <class _Cont, class _Ty>
+struct _Is_span_convertible_range
+    : is_convertible<remove_pointer_t<decltype(_STD data(_STD declval<_Cont&>()))> (*)[], _Ty (*)[]> {};
+
+// STRUCT TEMPLATE _Has_container_interface
+template <class, class = void>
+struct _Has_container_interface : false_type {};
+
+template <class _Cont>
+struct _Has_container_interface<_Cont,
+    void_t<decltype(_STD data(_STD declval<_Cont&>())), decltype(_STD size(_STD declval<_Cont&>()))>> : true_type {};
+
+// STRUCT TEMPLATE _Is_span_compatible_container
+// clang-format off
+template <class _Cont, class _Ty>
+struct _Is_span_compatible_container : conjunction<
+	negation<is_array<_Cont>>,
+	negation<_Is_span<remove_const_t<_Cont>>>,
+	negation<_Is_std_array<remove_const_t<_Cont>>>,
+	_Has_container_interface<_Cont>,
+	_Is_span_convertible_range<_Cont, _Ty>> {};

```

Comment:
There appear to be real tabs here. They should be 4-space indents.

Reply:
Fixed